File: Sorts/select_sort.go
package zysort

// select sort
func SelectSort(lst []int) {

	for i := 0; i < len(lst)-1; i++ {

		// pre-select the first to smallest
		min := i

		j := i

		for ; j < len(lst); j++ {
			if lst[min] > lst[j] {
				min = j
			}
		}

		if min != i {
			// swap it
			lst[min], lst[j] = lst[j], lst[min]
		}
	}
}

File: Sorts/select_sort.py
# -*- coding: utf-8 -*-


def select_sort(L):

    for i in range(len(L)):

        min = i
        
        for j in range(i+1, len(L)):
            if L[min] > L[j]:
                min = j

        L[min], L[i] = L[i], L[min]
